platform/highlight/highlight.go

```diff
@@ -0,0 +1,47 @@
+package highlight
+
+import (
+	"bytes"
+
+	"github.com/alecthomas/chroma"
+	"github.com/alecthomas/chroma/formatters/html"
+	"github.com/alecthomas/chroma/lexers"
+	"github.com/alecthomas/chroma/styles"
+)
+
+// Highlight the source using chroma in HTML format.
+// It will return the original content in plain text when it fails.
+func Highlight(source, lang, theme, linenr string) string {
+	lexer := lexers.Get(lang)
+	if lexer == nil {
+		lexer = lexers.Fallback
+	}
+	// coalesce runs of identical token types into a single token
+	lexer = chroma.Coalesce(lexer)
+
+	style := styles.Get(theme)
+	if style == nil {
+		// default theme is dracula
+		style = styles.Dracula
+	}
+
+	iterator, err := lexer.Tokenise(nil, source)
+	if err != nil {
+		return source
+	}
```

Comment:
error should be explicitly returned, change the function output signature to `(string, error)` please